### PR TITLE
🔒 Fix vulnerabilities

### LINKED ISSUE #101: ⚡️ 🔒  Fix package vulnerabilities
#### Describe your issue

We currently have a number of vulnerabilties with the runtime package (26 in fact), almost all of which are caused by the gulp- and gulptraum packages, which have vastly outdated dependencies.

Upgrading to gulp@4.0.0 would address most of these issues, but that would also entail some major refactoring of the gulptraum and gulptraum-typescript packages, as these are not yet ready to use that version.

Two of these vulnerabilities have a `critical` severity and 4 more are rated as `high`. With this in mind, I think taking some time to update these packages would be well worth the effort.

#### How can others reproduce the issue?

Run this in sequence:
- `npm install`
- `npm audit` **NOTE**: You'll best use `npm v6` for this, to get proper results.

You wil see a list of vulnerabilitites.

#### Possible solution

Fix the vulnerabilities in the gulptraum packages, release them with a new major version and then include that new version in all our packages.

On a sidenote:
There are also a couple of vulnerabilities with the `meta-npm` package, due to its usage of a vastly outdated version of `mocha`. 
A PR to address this is already open: https://github.com/mateodelnorte/meta-npm/pull/10
Once it has been merged and a new version published, we can update our own dependencies to that package.

However, these vulnerabilities will most likely not interfere with the runtime, as the only packages that use the `meta-npm` package are our own dev-internal meta packages, such as `process_engine_meta`, etc.

Once these vulnerabilitites have been addressed, we should be completely vulnerability-free. 🙂 

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by ElRaptorus**
Requires:
https://github.com/essential-projects/http_extension/pull/11
https://github.com/5minds/addict-ioc/pull/64
https://github.com/5minds/gulptraum/pull/35
https://github.com/5minds/gulptraum-typescript/pull/9

**Changes:**

1. Fix vulnerability issues
2. Upgrade to gulp v4
3. Add a package-lock that contains changes from a performed `npm audit fix` which fixed most of the vulnerability issues we were experiencing.

**NOTE**:
There are currently 8 low-level vulnerabilites left (compared to 26 vulnerabilties before, 2 of which were critical and 8 of them high), stating a possible prototype-pollution.
These are produced by `lodash` and `debug` and `npm audit fix` is unable to fix them (dunno why).
4 of them come from the package `vorpal`, used by gulptraum. This package uses `lodash`.
We can't throw that package out, because it is required for interacting with the gulp cli and producing comprehensive console logs. 

Concerning the other 4 vulnerabilities, they are produced by some usages of the `debug` package, but unfortunately the audit log is not really helpful in locating the occurences.

But I think, since these are all low-level vulnerabilities that don't really concern any really important package, I think we are fine for now.

**Also Note:**

Upgrading Gulp and addict-ioc has thus far only been done in this package, and the HttpExtension, since these the packages most heavily affected with vulnerabilities.
Uprading all the other packages will happen, once the following PRs for addict-ioc have been merged:
https://github.com/5minds/addict-ioc/pull/60
https://github.com/5minds/addict-ioc/pull/63

**Issues:**

Closes #101 

PR: #105

## How can others test the changes?

Everything should work as before.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,17 +84,17 @@
     "addict-ioc": "2.3.7",
     "bluebird": "3.5.2",
     "cross-env": "5.2.0",
-    "mocha": "5.0.0",
-    "mocha-jenkins-reporter": "0.3.12",
+    "mocha": "5.2.0",
+    "mocha-jenkins-reporter": "0.4.1",
     "moment": "2.22.2",
```
Comment (moellenbeck) [inline package.json]: Warum sind dev-Tools in den dependencies und nicht nur in den dev_dependencies?

File: package.json [hunk 2]
```diff
@@ -85,17 +84,17 @@
...
     "platform-folders": "0.2.7",
     "sequelize": "4.38.0",
-    "should": "13.1.3",
+    "should": "13.2.3",
```
Comment (moellenbeck) [inline package.json]: s.o.

File: package.json [hunk 3]
```diff
@@ -85,17 +84,17 @@
...
     "socket.io": "2.1.1",
     "socket.io-client": "2.1.1",
-    "sqlite3": "4.0.2",
+    "sqlite3": "4.0.3",
     "tsconfig": "7.0.0",
-    "typescript": "2.8.1",
+    "typescript": "3.1.3",
```
Comment (moellenbeck) [inline package.json]: s.o.

File: package.json (hunk 1 above)
Comment (moellenbeck) [reply package.json]: ich meine mocha, mocha-jenkins-reporter

**Comment by ElRaptorus** [comment]
@moellenbeck done

File: .gitignore [hunk 4]
```diff
@@ -14,10 +14,6 @@ sqlite_temp.json
 metrics
 logs
 
-# lock-files
-package-lock.json
-yarn.lock
-
```
Comment (robpal94) [inline .gitignore]: Should we commit all our `lock` files now or does this only apply for this repository?

**Pull request merged**